### PR TITLE
Expand browsers support

### PR DESCRIPTION
**Pull request by yifanyang**
Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,11 +27,21 @@ addons:
...
-  on:
-    repo: firebase/firebase-js-sdk
-    branch: master
+jobs:
+  include:
+    - script: xvfb-run yarn test
+      after_success: yarn test:coverage
+      env: '#comment: yarn test'
+    - script: xvfb-run yarn test:saucelabs
+      env: '#comment: yarn test:saucelabs'
```
Comment (jshcrowthe) [inline .travis.yml]: So I think we want to just completely disable this stage in the event that it is not a push build, I just realized that we run the same suite twice differently in two browsers and that seems unnecessary.

See: https://docs.travis-ci.com/user/conditional-builds-stages-jobs/ for an easy way of doing this.

File: config/karma.saucelabs.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const glob = require('glob');
+const karma = require('karma');
+const path = require('path');
+const karmaBase = require('./karma.base');
+
+/** Tests in these packages are exluded due to flakiness or long run time. */
```
Comment (jshcrowthe) [inline config/karma.saucelabs.js]: nit: exluded -> excluded

File: config/karma.saucelabs.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    sauceLabs: {
+      tunnelIdentifier: process.env.TRAVIS_JOB_NUMBER,
+      username: process.env.SAUCE_USERNAME,
+      accessKey: process.env.SAUCE_ACCESS_KEY,
+      startConnect: true
+    }
+  });
+
+  /* https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions */
+  if (process.env.TRAVIS_PULL_REQUEST != 'false') {
```
Comment (jshcrowthe) [inline config/karma.saucelabs.js]: Per comment above: 

Let's go ahead and run these tests only during the push builds and preserve the resource if we can (see: https://docs.travis-ci.com/user/conditional-builds-stages-jobs/)

We can probably remove this block.

File: config/karma.saucelabs.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+};
+
+module.exports = function(config) {
+  const karmaConfig = Object.assign({}, karmaBase, {
+    basePath: '../',
+
+    files: getTestFiles(),
+
+    client: Object.assign({}, karmaBase.client, {
+      firestoreSettings: {
```
Comment (jshcrowthe) [inline config/karma.saucelabs.js]: Can you leave a comment explaining why this is needed?

File: .travis.yml (hunk 1 above)
Comment (yifanyang) [reply .travis.yml]: Done.

I'm using env as a description for these two Travis jobs. It turns out that for if condition to work, there has to be a '=' in the env statement.

File: config/karma.saucelabs.js (hunk 2 above)
Comment (yifanyang) [reply config/karma.saucelabs.js]: Done.

File: config/karma.saucelabs.js (hunk 3 above)
Comment (yifanyang) [reply config/karma.saucelabs.js]: Done.

File: config/karma.saucelabs.js (hunk 4 above)
Comment (yifanyang) [reply config/karma.saucelabs.js]: Removed.

This was copied from karma config from packages/firestore. But we don't actually need it, as firestore tests are excluded for long running time right now. We still want to enable firestore tests eventually, and will figure out what exactly need to be done later.

Comment (yifanyang) [reply config/karma.saucelabs.js]: Yes. Well, I guess we could added it back the day we enabled the firestore tests.

**Pull request merged**